smartsim/_core/launcher_/slurm/slurm_launcher.py

```diff
@@ -0,0 +1,334 @@
+            if status is not None
+        }
+        slurm_ids_to_status = collections.ChainMap(has_overwrite, fetched_status)
+        return {id_: slurm_ids_to_status[id_to_info[id_].slurm_id] for id_ in ids}
+
+    @staticmethod
+    def _get_status(
+        id_: parser.StepID, *ids: parser.StepID
+    ) -> dict[parser.StepID, JobStatus]:
+        """Given a collection of step ids, interogate slurm for the status of
+        the steps.
+
+        :param id_: The first step id to get the status for.
+        :param ids: Any additional step ids to get the status for.
+        :returns: A mapping of step ids to statuses
+        """
+        ids = (id_,) + ids
+        # >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+        # TODO: Really don't like the implied sacct format and required
+        #       order to call these fns in order to get the status
+        # -------------------------------------------------------------------------
+        out, _ = commands.sacct(
+            ["--noheader", "-p", "-b", "--jobs", ",".join(ids)], raise_on_err=True
+        )
+        stats = ((id_, parser.parse_sacct(out, id_)[0]) for id_ in ids)
+        # <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+        to_ss_stat = {
+            "RUNNING": JobStatus.RUNNING,
+            "CONFIGURING": JobStatus.RUNNING,
+            "STAGE_OUT": JobStatus.RUNNING,
+            "COMPLETED": JobStatus.COMPLETED,
+            "DEADLINE": JobStatus.COMPLETED,
+            "TIMEOUT": JobStatus.COMPLETED,
+            "BOOT_FAIL": JobStatus.FAILED,
+            "FAILED": JobStatus.FAILED,
+            "NODE_FAIL": JobStatus.FAILED,
+            "OUT_OF_MEMORY": JobStatus.FAILED,
+            "CANCELLED": JobStatus.CANCELLED,
+            "CANCELLED+": JobStatus.CANCELLED,
+            "REVOKED": JobStatus.CANCELLED,
+            "PENDING": JobStatus.PAUSED,
+            "PREEMPTED": JobStatus.PAUSED,
+            "RESV_DEL_HOLD": JobStatus.PAUSED,
+            "REQUEUE_FED": JobStatus.PAUSED,
+            "REQUEUE_HOLD": JobStatus.PAUSED,
+            "REQUEUED": JobStatus.PAUSED,
+            "RESIZING": JobStatus.PAUSED,
+            "SIGNALING": JobStatus.PAUSED,
+            "SPECIAL_EXIT": JobStatus.PAUSED,
+            "STOPPED": JobStatus.PAUSED,
+            "SUSPENDED": JobStatus.PAUSED,
+        }
+        return {id_: to_ss_stat.get(stat, JobStatus.UNKNOWN) for id_, stat in stats}
+
+    def stop_jobs(self, *ids: LaunchedJobID) -> t.Mapping[LaunchedJobID, JobStatus]:
+        """Take a collection of job ids and kill the corresponding processes
+        started by the slurm launcher.
+
+        :param ids: The ids of the launched jobs to stop.
+        :returns: A mapping of ids to their reported status after attempting to
+            stop them.
+        """
+        slurm_infos = tuple(map(self._get_slurm_info_from_job_id, ids))
+        for info in slurm_infos:
+            self._stop_job(info)
```

Comment:
Should we do these asynchronously to avoid waiting for each job to stop?

Reply:
Makes sense to me! Now running each `scancel` in a new thread!